### PR TITLE
Don't load plugins inside Scheduling loop

### LINKED ISSUE #13099: Unable to start scheduler after stopped
**Apache Airflow version**: 2.0.0rc3

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`):

**Environment**:

- **Cloud provider or hardware configuration**: Linux
- **OS** (e.g. from /etc/os-release): Ubuntu
- **Kernel** (e.g. `uname -a`): 
- **Install tools**:
- **Others**:

**What happened**:

After shutting down the scheduler, while tasks were in running state, trying to restart the scheduler results in pk violations..

```[2020-12-15 22:43:29,673] {scheduler_job.py:1293} ERROR - Exception when executing SchedulerJob._run_scheduler_loop
Traceback (most recent call last):
  File "/home/jcoder/git/airflow_2.0/pyenv/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
    cursor, statement, parameters, context
  File "/home/jcoder/git/airflow_2.0/pyenv/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "dag_run_dag_id_run_id_key"
DETAIL:  Key (dag_id, run_id)=(example_task_group, scheduled__2020-12-14T04:31:00+00:00) already exists.
```

**What you expected to happen**:

Scheduler restarts and picks up where it left off.

**How to reproduce it**:

Set example dag ( I used task_group) to schedule_interval `* * * * *` and start the scheduler and let it run for a few minutes. 
Shut down the scheduler
Attempt to restart the scheduler

**Anything else we need to know**:
I came across this doing testing using the LocalExecutor in a virtual env. If no else is able to reproduce it, I'll try again in a clean virtual env.

### PR DESCRIPTION
**Pull request by kaxil**
closes https://github.com/apache/airflow/issues/13099

Mutually exclusive with https://github.com/apache/airflow/pull/13932

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/jobs/scheduler_job.py [hunk 1]
```diff
@@ -1466,6 +1466,8 @@ def _do_scheduling(self, session) -> int:
         :return: Number of TIs enqueued in this iteration
         :rtype: int
         """
+        os.environ["AIRFLOW_IN_SCHEDULING_LOOP"] = 'True'
+
```
Comment (kaxil) [inline airflow/jobs/scheduler_job.py:1470]: Once we decided on which PR is better, I will add code comments + tests

**Comment by kaxil** [comment]
@ephraimbuddy @jhtimmins Can you take a look and review too please

File: airflow/serialization/serialized_objects.py [hunk 2]
```diff
@@ -407,35 +408,39 @@ def serialize_operator(cls, op: BaseOperator) -> Dict[str, Any]:
...
+        # We don't want to load Extra Operator links in Scheduler
+        if load_op_links:  # pylint: disable=too-many-nested-blocks
+            from airflow import plugins_manager
+
+            plugins_manager.initialize_extra_operators_links_plugins()
+
+            if plugins_manager.operator_extra_links is None:
+                raise AirflowException("Can not load plugins")
+
+            for ope in plugins_manager.operator_extra_links:
```
Comment (jhtimmins) [inline airflow/serialization/serialized_objects.py:430]: Some of these variable names are ambiguous. Particularly `ope`

Comment (kaxil) [reply airflow/serialization/serialized_objects.py:430]: They indeed are but this just indents the code and does not change it, use https://github.com/apache/airflow/pull/13931/files?diff=split&w=1 to hide whitespace changes

**Comment by kaxil** [comment]
Closing this in favor of https://github.com/apache/airflow/pull/13932

**Pull request closed without merging**